### ISSUE TITLE
Prompted to Reload Project after Adding Pre or Post Deploy Script

### ISSUE THREAD

**Issue by chlafreniere** [microsoft/azuredatastudio#22530]
(after reloading ADS I can no longer repro 😞)

<img width="925" alt="image" src="https://user-images.githubusercontent.com/40371649/228683217-4733fee0-2b74-4ec5-bb49-9a73f1c995fa.png">

Previously, when adding pre or post deploy script, I saw the above message. Will keep trying to see if I can get a repro.

Mac (x64), SDK-style sqlproj

**Comment by Charles-Gagnon** [comment]
Ok, I have a consistent initial repro: 

1. Reload ADS with sqlproj in workspace
2. Right click on project and click "Edit .sqlproj"
3. Now do anything that modifies the project, such as adding a folder

Doing these steps I'm seeing the toast notification pop up for that and any subsequent modifications.

**Comment by kisantia** [comment]
yes for that case it's expected for the notification to popup since there's a file watcher when the .sqlproj is opened using the "Edit .sqlproj File" command that only gets disposed when the file is closed. This was intended for when users to get prompted to reload the project if they edit their .sqlproj, with the expectation that they'll close the file when they're done editing. 

Think this is too much and should be removed? The user could also use the refresh button to reload their project if we got rid of this file watcher and notification

**Comment by Charles-Gagnon** [comment]
Is there a reason you wouldn't refresh any time you detected a change automatically?

**Comment by kisantia** [comment]
Hmm not sure why we didn't do that originally. Maybe it was to make it more similar to SSDT, which detects if the .sqlproj has been changed and prompts the user if they want to reload or not. If not, then the next save will overwrite the .sqlproj with the project state in SSDT.
